### ISSUE TITLE
ui edge case: Wrong empty state and filter sidebar after canceling "Add new label"

### ISSUE THREAD

**Issue by mikermcneil** [kolide/fleet#2326]
#### Step 1: From here, choose "Add new label"
![image](https://user-images.githubusercontent.com/618009/95639118-ea5a7d80-0a5c-11eb-994e-5338191fa28a.png)

#### Step 2: From here, choose "Cancel"
![image](https://user-images.githubusercontent.com/618009/95639129-f5151280-0a5c-11eb-9c27-391d38d4f039.png)

### What did you expect to see?
![image](https://user-images.githubusercontent.com/618009/95639193-268dde00-0a5d-11eb-9112-af69d36a504e.png)

### What did you see instead?
![image](https://user-images.githubusercontent.com/618009/95639175-1a098580-0a5d-11eb-88b6-039a034dd19c.png)

### In summary
Looks like the issue is that the filter stays applied but the right sidebar doesn't draw it as if it's applied, and the wrong empty state is being displayed.
